components/script_bindings/domstring.rs

```diff
@@ -675,6 +701,50 @@ impl From<Cow<'_, str>> for DOMString {
     }
 }
 
+/// Use this to match &str against lazydomstring efficiently.
+/// You are only allowed to match ascii strings otherwise this macro will
+/// lead to wrong results.
+/// ```ignore
+/// let s = DOMString::from_string(String::from("test"));
+/// let value = match_domstring!(s, 0,
+/// "test1" => 1,
+/// "test2" => 2,
+/// "test" => 3,);
+/// assert_eq!(value, 3);
+/// ```
+#[macro_export]
+macro_rules! match_domstring_ascii {
+    ( $input:expr, $default:expr,
+        $(
+            $p: expr  => $then: expr
```

Comment:
If we use `$p: pat` then we don't need the $default argument at all and we can keep the _ branches in the callsites.